### DISCUSSION TITLE
Option for Combobox results to include tags

### DISCUSSION THREAD

**Discussion by nicedisguise** [twilio-labs/paste#1647] [General] [answered]
Hi there,
I'm currently working on this flow:
https://www.figma.com/file/CBFtsLBLkdwcDQ1iw3VXU5/Admin-Experience-Conversations?node-id=369%3A78099

I wondered if it was possible to include (a) a tag and (b) disabled results in the dropdown?

FYI - The reason for the disabled results being shown is because we can't not show certain numbers due to some technical constraints when searching/filtering numbers. We _could_ make them selectable but that would lead to a poor UX where an error would occur after the user has clicked the 'submit' button at the bottom of the page. We'd rather be up front with what you can and can't select.

**Comment by gloriliale** [comment, marked as answer]
Hi Duncan, great question!

Putting tags (or ~~pills~~ badges, as we also refer to them) inside of the combobox options is definitely possible. Priyanka has done something similar, so I'd recommend connecting with her to see how it was constructed.

As far as creating disabled options, I wanted to first ask- is there a reason you'd rather go that route rather than keep the numbers from becoming options in the combobox in the first place?

**Reply by ssonajalg** [reply]
Hi!
We are working on registering account's phone numbers into Flex. To do so, we need to merge 2 datasources - Phone Numbers API (for all phone numbers), and Flex Flows API (for already registered numbers). For some accounts, both will contain thousands of entries. It's probably obvious that we can't just load both whole resultSets into memory while rendering this page - we need to paginate. But the first page of Phone Numbers can easily contain 50-out-of-50 registered numbers. Now if we'd want to filter down to first page of _unregistered_ numbers, we'd potentially be scrolling the Phone Numbers API through thousands of numbers, plus making expensive extra queries to the other API to figure out which numbers are unregisted.

Long story short - with current architecture, getting the first page of _unregistered_ numbers is technically unfeasible/unsafe/wouldn't perform. We just need to show whatever it is on the first page, registerd or not. But already registered numbers cannot be selected.

**Reply by gloriliale** [reply]
Hi @ssonajalg and @nicedisguise!

Ah, I see why this is getting so complicated. Sadly it isn't possible to disable options inside of Combobox.

If there's no way around this technical limitation, I wonder if this part of the flow could be pulled out into its own page. It could show a loading animation and some text like “We are pulling a list of available phone numbers for you to choose from. This could take a few minutes.”
